apps/meteor/tests/e2e/omnichannel/omnichannel-business-hours.spec.ts

```diff
@@ -0,0 +1,144 @@
+        await expect(poOmnichannelBusinessHours.confirmDeleteModal).toBeVisible();
+        await poOmnichannelBusinessHours.btnConfirmDeleteModal.click();
+        await expect(poOmnichannelBusinessHours.confirmDeleteModal).not.toBeVisible();
+      });
+    });
+
+    await test.step('expect business hours to have been deleted', async () => {
+      await poOmnichannelBusinessHours.search(BHName);
+      await expect(poOmnichannelBusinessHours.findRowByName(BHName)).not.toBeVisible();
+    });
+  });
+
+  test('OC - Business hours - Edit BH departments', async ({ api, page }) => {
+    const BHName = faker.string.uuid();
+
+    await test.step('expect to create new businessHours', async () => {
+			const createBH = await createBusinessHour(api, {
+        id: '33',
+				name: BHName,
+				departments: [department.data._id],
+			});
+
+      expect(createBH.status()).toBe(200);
+		});
+
+    await page.goto('/omnichannel');
+		await poOmnichannelBusinessHours.sidenav.linkBusinessHours.click();
+
+    const department2 = await createDepartment(api);
+
+    await test.step('expect to add business hours departments', async () => {
+			await poOmnichannelBusinessHours.search(BHName);
+			await poOmnichannelBusinessHours.findRowByName(BHName).click();
+			await poOmnichannelBusinessHours.selectDepartment({ name: department2.data.name, _id: department2.data._id });
+			await poOmnichannelBusinessHours.btnSave.click();
+		});
+
+		await test.step('expect department to be in the chosen departments list', async () => {
+			await poOmnichannelBusinessHours.search(BHName);
+			await poOmnichannelBusinessHours.findRowByName(BHName).click();
+			await expect(page.getByRole('option', { name: department2.data.name })).toBeVisible();
+      await poOmnichannelBusinessHours.btnBack.click();
+		});
+
+    await test.step('expect to remove business hours departments', async () => {
+			await poOmnichannelBusinessHours.search(BHName);
+			await poOmnichannelBusinessHours.findRowByName(BHName).click();
+			await poOmnichannelBusinessHours.selectDepartment({ name: department2.data.name, _id: department2.data._id });
+			await poOmnichannelBusinessHours.btnSave.click();
+		});
+
+		await test.step('expect department to not be in the chosen departments list', async () => {
+			await poOmnichannelBusinessHours.search(BHName);
+			await poOmnichannelBusinessHours.findRowByName(BHName).click();
+			await expect(page.getByRole('option', { name: department2.data.name })).toBeHidden();
+      await poOmnichannelBusinessHours.btnBack.click();
+		});
+
+    await test.step('expect delete business hours', async () => {
+      await poOmnichannelBusinessHours.btnDeleteByName(BHName).click();
+      await expect(poOmnichannelBusinessHours.confirmDeleteModal).toBeVisible();
+      await poOmnichannelBusinessHours.btnConfirmDeleteModal.click();
+      await expect(poOmnichannelBusinessHours.confirmDeleteModal).not.toBeVisible();
+    });
+
+		await department2.delete();
```

Comment:
Can we do the creation and removal of this department2 in a beforeEach/AfterEach statement? It will ensure that it is properly removed even if the test fails. In the current implementation, if anything goes wrong before the department2.delete statement, it will not run and leave undesired data behind.

You can specify on what test you want the code to run by using an `if` with the [testInfo](https://playwright.dev/docs/api/class-testinfo) object